plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -610,8 +618,8 @@ func (r *ReferenceManager) ensureSecretBindingReferences(ctx context.Context, at
 
 func (r *ReferenceManager) ensureShootReferences(ctx context.Context, attributes admission.Attributes, oldShoot, shoot *core.Shoot) error {
 	if !equality.Semantic.DeepEqual(oldShoot.Spec.CloudProfileName, shoot.Spec.CloudProfileName) {
-		if _, err := r.cloudProfileLister.Get(shoot.Spec.CloudProfileName); err != nil {
-			return err
+		if _, err := utils.GetCloudProfile(r.cloudProfileLister, r.namespacedCloudProfileLister, shoot.Spec.CloudProfileName, shoot.Namespace); err != nil {
+			return fmt.Errorf("could not find referenced (private) cloud profile when ensuring shoot references: %+v", err.Error())
```

Comment:
```suggestion
			return fmt.Errorf("could not find referenced cloud profile when ensuring shoot references: %w", err)
```

Since there is no term "private".